Core/Src/leds/rgb_led_handler.c

```diff
@@ -40,31 +48,29 @@ void rgb_led(void* argument) {
         ECU_ENABLE_BREAKPOINT_DEBUG();
 
         // espera RTD ser setado ou timeout estourar
-        switch (
-            osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL, RGB_BLINK_DELAY)) {
+        switch (osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL, RGB_TIMEOUT)) {
 
             // caso timeout estore vai piscar o led, indicando que ta fora do RTD
-            case osErrorTimeout:
-                write_rgb_color(get_rgb_color(message.color));
-                blink_rgb(RGB_BLINK_DELAY);
-                break;
+            case osErrorTimeout: write_pattern(message, RGB_BLINK_200_DELAY); break;
 
             default:
                 switch (message.control) {
                     case FIXED:
-                        for (;;) {
-                            write_rgb_color(get_rgb_color(message.color));
-                            osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL,
-                                              osWaitForever);
-                            if (message.control == BLINK200) {
-                                break;
+                        if (message.sizeOfPattern > 1) {
+                            for (;;) {
+                                write_pattern(message, RGB_BLINK_1000_DELAY);
+                                if (osMessageQueueGetCount(q_rgb_led_messageHandle) > 0) {
+                                    break;
+                                }
                             }
```

Comment:
De novo muita identação aqui que da pra reduzir e melhorar bem a legibilidade. Cria no lugar no `messageWait()` essa `is_zero_messages_available()`. 
```suggestion
                    if (message.sizeOfPattern > 1) {
                        while (is_zero_messages_available(q_rgb_led_messageHandle)) {
                            write_pattern(message, RGB_BLINK_1000_DELAY);
                        }
```

Reply:
fiz essa fução is_zero_message_available assim:

int is_zero_message_available(osMessageQueueId_t q_id) {
    return osMessageQueueGetCount(q_rgb_led_messageHandle);
}

Reply:
o nome  is_zero_message_available é assim para parecer uma pergunta. 

Tem zero mensagens disponiveis? 
Se tiver zero mensagens faça tal

Então ele tem que retornar true para zero mensagens